src/Transport/Topology/MetaModel/NamespaceBundleConfigurations.cs

```diff
@@ -0,0 +1,44 @@
+﻿namespace NServiceBus.AzureServiceBus.Topology.MetaModel
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    class NamespaceBundleConfigurations : IEnumerable<NamespaceBundleInfo>
+    {
+        List<NamespaceBundleInfo> namespaceBundles;
+
+        public NamespaceBundleConfigurations()
+        {
+            namespaceBundles = new List<NamespaceBundleInfo>();
+        }
+
+        public void Add(string namespaceAlias, int numberOfTopicsInBundle)
+        {
+            var bundleInfo = namespaceBundles.SingleOrDefault(x => x.Alias.Equals(namespaceAlias, StringComparison.OrdinalIgnoreCase));
+            if (bundleInfo != null && bundleInfo.NumberOfTopicsInBundle < numberOfTopicsInBundle)
+            {
+                throw new Exception($"Duplicate namespace alias `{namespaceAlias}` was added to the {nameof(NamespaceBundleConfigurations)}. Namespace aliases should be unique.");
+            }
+
+            namespaceBundles.Add(new NamespaceBundleInfo(namespaceAlias, numberOfTopicsInBundle));
+        }
+
+        public int GetNumberOfTopicInBundle(string namespaceAlias)
+        {
+            var selected = namespaceBundles.SingleOrDefault(x => x.Alias.Equals(namespaceAlias, StringComparison.OrdinalIgnoreCase));
+            return selected?.NumberOfTopicsInBundle ?? 0;
```

Comment:
The caller doesn't "fix it". 

For the called class, `NamespaceBundleConfigurations` to know what the most recent setting value is, I would have to pass in the number of configured topics in bundle. That is not know until settings are locked (yes, I do not want to pass the whole Settings object since we're trying to avoid that). Finding when that happens is far from trivial.

What the method does is absolutely legitimate. If this method cannot find the namespace alias (which is probably odd on it's own since aliases registered are the ones that user has already provided), the value returned is compared with the setting (either 2, which is configured by default, or whatever user specified).